protocol/messenger_test.go

```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
+
+		chat.UnviewedMessagesCount = 1
+		chat.UnviewedMentionsCount = 0
+		chat.Highlight = true
+
+		err := s.m.SaveChat(chat)
+		s.Require().NoError(err)
+
+		inputMessage1 := buildTestMessage(*chat)
+		inputMessage1.ID = "1"
+		inputMessage1.Seen = false
+		inputMessage1.Mentioned = false
+
+		err = s.m.SaveMessages([]*common.Message{inputMessage1})
+		s.Require().NoError(err)
+
+		chats := s.m.Chats()
+
+		for idx := range chats {
+			if chats[idx].ID == chat.ID {
+				s.Require().Equal(uint(1), chats[idx].UnviewedMessagesCount)
+				s.Require().Equal(uint(0), chats[idx].UnviewedMentionsCount)
+				s.Require().Equal(true, chats[idx].Highlight)
+			}
+		}
+
+		count, countWithMentions, err := s.m.MarkMessageAsUnread(chat.ID, inputMessage1.ID)
+		s.Require().NoError(err)
+		s.Require().Equal(uint64(1), count)
+		s.Require().Equal(uint64(0), countWithMentions)
+
+		chats = s.m.Chats()
+
+		for idx := range chats {
+			if chats[idx].ID == chat.ID {
+				s.Require().Equal(uint(1), chats[idx].UnviewedMessagesCount)
+				s.Require().Equal(uint(0), chats[idx].UnviewedMentionsCount)
+				s.Require().Equal(false, chats[idx].Highlight)
+			}
+		}
+	})
+
+	s.T().Run("Test marking a message as unread in a chat does not impact other chat", func(t *testing.T) {
+		chat1 := CreatePublicChat("test-chat-1", s.m.transport)
+		err := s.m.SaveChat(chat1)
+		s.Require().NoError(err)
+
+		chat2 := CreatePublicChat("test-chat-2", s.m.transport)
+		err = s.m.SaveChat(chat2)
+		s.Require().NoError(err)
+
+		chat1.UnviewedMessagesCount = 2
+		chat1.UnviewedMentionsCount = 0
+		chat1.Highlight = true
+
+		inputMessage1 := buildTestMessage(*chat1)
+		inputMessage1.ID = "1"
+		inputMessage1.Seen = false
+		inputMessage1.Mentioned = false
+
+		time.Sleep(1000 * time.Millisecond)
+
+		inputMessage2 := buildTestMessage(*chat1)
+		inputMessage2.ID = "2"
+		inputMessage2.Seen = false
+		inputMessage2.Mentioned = false
+
+		err = s.m.SaveMessages([]*common.Message{inputMessage1, inputMessage2})
+		s.Require().NoError(err)
+
+		chat2.UnviewedMessagesCount = 1
+		chat2.UnviewedMentionsCount = 0
+		chat2.Highlight = true
+
+		time.Sleep(1000 * time.Millisecond)
+
+		inputMessage3 := buildTestMessage(*chat2)
+		inputMessage3.ID = "3"
+		inputMessage3.Seen = false
+		inputMessage3.Mentioned = false
+
+		err = s.m.SaveMessages([]*common.Message{inputMessage3})
+		s.Require().NoError(err)
+
+		count, countWithMentions, err := s.m.MarkMessageAsUnread(chat1.ID, inputMessage1.ID)
+
+		s.Require().NoError(err)
+		s.Require().Equal(uint(1), chat2.UnviewedMessagesCount)
+		s.Require().Equal(uint64(2), count)
+		s.Require().Equal(uint64(0), countWithMentions)
+
+		chats := s.m.Chats()
+
+		for idx := range chats {
+			if chats[idx].ID == chat1.ID {
+				s.Require().Equal(uint(2), chats[idx].UnviewedMessagesCount)
+				s.Require().Equal(uint(0), chats[idx].UnviewedMentionsCount)
+				s.Require().Equal(false, chats[idx].Highlight)
+			} else if chats[idx].ID == chat2.ID {
```

Comment:
This is not enough here.
Since you don't update chat2 in `messenger.allChats`, the value will surely remain the same.
We should check that the `Seen` property for the message in the database in correct.

Reply:
that's right, updated all tests to account for the same, meaning testing the .Seen property.